nima/webserver/webserver/src/main/java/io/helidon/nima/webserver/task/spi/HelidonTaskExecutor.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.webserver.task.spi;
+
+import java.io.Closeable;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A simplified {@link java.util.concurrent.ExecutorService} that can execute
+ * {@link InterruptableTask}s and can be efficiently terminated. A thread that is
+ * waiting to read on an open connection cannot be efficiently stopped. This
+ * executor will query the thread and interrupt it if possible. It is important
+ * to efficiently shut down the Nima webserver efficiently certain environments.
```

Comment:
double `efficiently` in the last sentence.

Reply:
I think this package should not exist. I would either move it to `webserver` directly, or to `common.configurable` to live with our `ThreadPoolSupplier`. 
This may even deserve a new module in `common`, as I can imagine other components using the same (even if not webserver related).

Reply:
Also this is not really `spi`

Reply:
Moved to `webserver` since we don't yet have a use case for `common` AFAIK. We can revisit this later.